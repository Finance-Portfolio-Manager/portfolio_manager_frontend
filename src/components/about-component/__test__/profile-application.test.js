import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import ProfileComponent from './../profile-application';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const wrapper = shallow(<ProfileComponent></ProfileComponent>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<ProfileComponent></ProfileComponent>).toJSON();
    expect(tree).toMatchSnapshot();
})

test("card body renders", ()=>{
    expect(wrapper.find("CardBody").length).toBe(1);
})

test("title renders", ()=>{
    expect(wrapper.find("p").first().text()).toContain("Meet Our Team");
})

test("3 team buttons render", ()=>{
    expect(wrapper.find("Button").length).toBe(3);
})

test("doesn't crash on button click", ()=>{
    const clickFn = jest.fn();
    const component = shallow(<ProfileComponent></ProfileComponent>);
    const firstButton = component.find("Button").first();
    console.log(firstButton.debug());
    expect(firstButton.simulate('click', {target: {id: "team1"}})).toBeTruthy();
})