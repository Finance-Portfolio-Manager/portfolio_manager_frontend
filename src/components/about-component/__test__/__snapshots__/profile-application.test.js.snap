// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
>
  <p
    className="d-flex flex-column w-50 meetTeam justify-content-center mx-auto lead fw-bolder text-white primary-text"
    id="meet"
  >
    Meet Our Team
  </p>
  <hr
    className="text-white"
  />
  <section
    id="profiles "
  >
    <div
      className="card border-0 d-flex flex-column captainProfile justify-content-center mx-auto my-4"
      id="individual"
    >
      <div>
        <a
          className="text-decoration-none text-white "
          href="https://www.linkedin.com/in/carolynhrehm/"
          target="_blank"
        >
          <img
            className="headImage"
            src="https://media-exp1.licdn.com/dms/image/C4E03AQHjrLZPKLi7WQ/profile-displayphoto-shrink_400_400/0/1517480368667?e=1634169600&v=beta&t=l4kjtEM3n4__wF9UOwPbRO9hXXJcdjKxOPQG5IDpVmY"
          />
        </a>
      </div>
      <hr />
      <div
        className="card-body"
      >
        <p
          className="primary-text"
        >
          Carolyn Rehm
        </p>
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="teams m-3 col"
      >
        <button
          className="team-button border-0 btn btn-primary"
          disabled={false}
          id="team1"
          onClick={[Function]}
          type="button"
        >
          Team1
        </button>
      </div>
      <div
        className="teams m-3 col"
      >
        <button
          className="team-button border-0 btn btn-primary"
          disabled={false}
          id="team2"
          onClick={[Function]}
          type="button"
        >
          Team2
        </button>
      </div>
      <div
        className="teams m-3 col"
      >
        <button
          className="team-button border-0 btn btn-primary"
          disabled={false}
          id="team3"
          onClick={[Function]}
          type="button"
        >
          Team3
        </button>
      </div>
    </div>
    <br />
    <div
      className="row"
    />
  </section>
</div>
`;
